src/cmd/compile/internal/walk/compare.go

```diff
@@ -5,16 +5,35 @@
 package walk
 
 import (
-	"go/constant"
-
 	"cmd/compile/internal/base"
 	"cmd/compile/internal/ir"
 	"cmd/compile/internal/reflectdata"
 	"cmd/compile/internal/ssagen"
 	"cmd/compile/internal/typecheck"
 	"cmd/compile/internal/types"
+	"encoding/binary"
+	"fmt"
+	"go/constant"
+	"hash/fnv"
+	"io"
 )
 
+func fakePC(n ir.Node) ir.Node {
+	// In order to get deterministic IDs, we include the package path, file index, line number, column number
+	// in the calculation of the fakePC for the IR node.
+	hash := fnv.New32()
+	// We ignore the errors here because the `io.Writer` in the `hash.Hash` interface never returns an error.
+	_, _ = io.WriteString(hash, base.Ctxt.Pkgpath)
+	_ = binary.Write(hash, binary.LittleEndian, n.Pos().FileIndex())
+	_ = binary.Write(hash, binary.LittleEndian, int64(n.Pos().Line()))
+	_ = binary.Write(hash, binary.LittleEndian, int64(n.Pos().Col()))
+	// We also include the string representation of the node to distinguish autogenerated expression since
+	// those get the same `src.XPos`
+	_, _ = io.WriteString(hash, fmt.Sprintf("%v", n))
```

Comment:
These nodes seem to have `AutoTmp()` set and you may be able to get the name from `.Name()` directly. We should try not to rely on `"%v"`.

Reply:
I see, it's probably better than the alternatives then. Do we still have to hash in the other fields or are they also part of the string representation?